modules/core/04-channel/keeper/packet_test.go

```diff
@@ -713,6 +733,18 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 			suite.Require().NoError(err)
 			channelCap = suite.chainA.GetChannelCapability(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
 		}, false},
+		{"channel in ackupgrade flush status not in flush", func() {
+			expError = types.ErrInvalidChannelState
+
+			suite.coordinator.Setup(path)
+			packet = types.NewPacket(ibctesting.MockPacketData, 1, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+
+			channel := path.EndpointA.GetChannel()
+			channel.State = types.ACKUPGRADE
+			channel.FlushStatus = types.NOTINFLUSH
+			path.EndpointA.SetChannel(channel)
```

Comment:
Ideally we would use the existing handlers to enter into valid state. Technically it isn't possible to end up in `NOTINFLUSH` and `ACKUPGRADE`, the flush status should be `FLUSHCOMPLETE`

Reply:
does it make test to use the handlers to enter an valid state (`ACKUPGRADE`) only to yank the channel, alter flush status and then ack the packet?

Reply:
disregard previous comment. moved this to ack with handlers and manually set flush status to flush complete. linking #3928 here too.